upgrade_guide.md

```diff
@@ -0,0 +1,374 @@
+## Class names
+All classes have been namespaced. The mostly used classes are now as follows:
+ 
+- `Algolia::Client` -> `Algolia::Search::Client`
+- `Algolia::Index` -> `Algolia::Search::Index`
+- `Algolia::AccountClient` -> `Algolia::Account::Client`
+- `Algolia::Analytics` -> `Algolia::Analytics::Client`
+- `Algolia::Insights` -> `Algolia::Insights::Client`
+
+## Initialize the client
+There's a slight change in how you initialize the client.
+```ruby
+client = Algolia::Search::Client.create('APP_ID', 'API_KEY')
+# OR
+search_config = Algolia::Search::Config.new(app_id: app_id, api_key: api_key)
+client = Algolia::Search::Client.new(search_config)
+```
+
+## Initialize an index
+The way you instantiate an Index from the client did not change.
+```ruby
+client.init_index('index_name')
+```
+
+## Search parameters and request options
+The search parameters and request options are still optional, but they are combined into a single hash instead of two. 
+For example:
+```ruby
+opts = {
+  :headers => {
+    'X-Algolia-UserToken': 'user123'
+  },
+  :params => {
+    hitsPerPage: 50
+  }
+}
+index.search('query', opts)
+```
+
+## Methods
```

Comment:
You can get some inspiration from this table https://www.algolia.com/doc/api-client/getting-started/upgrade-guides/php/#list-of-method-signature-changes and maybe write the before/after signatures